library/std/src/sys/uefi/fs.rs

```diff
@@ -0,0 +1,1022 @@
+//! File System functionality for UEFI
+
+use crate::ffi::{OsStr, OsString};
+use crate::fmt;
+use crate::hash::Hash;
+use crate::io::{self, BorrowedCursor, IoSlice, IoSliceMut, SeekFrom};
+use crate::path::{Path, PathBuf};
+use crate::sys::time::SystemTime;
+use crate::sys::unsupported;
+use r_efi::protocols::file;
+
+pub struct File {
+    ptr: uefi_fs::FileProtocol,
+    path: PathBuf,
```

Comment:
Maybe it would be possible to use `Open("..")` along with `GetInfo()` to reverse through the ancestors of the file and construct the path that way instead of storing it in the structure? That would also fix situations in which the file is renamed or moved. Additionally, it is only used in the debug formatter, so it might not be worth the size increase.

Reply:
I do have some concerns about doing that: 
1. How will we find the Device Prefix? 
2. UEFI File Protocol, is well not the fastest. So I'm not sure if the overhead of opening multiple files and using `GetInfo` on them will be worth it

Reply:
By the way, displaying the file path on debug, while useful, is not really necessary. Most of the smaller systems supported by `std` do not do so.

Reply:
> By the way, displaying the file path on debug, while useful, is not really necessary. Most of the smaller systems supported by `std` do not do so.

So what would you suggest to display in Debug? is just the file name sufficient?